### PR TITLE
CRM 33 - ExceptionHandler

### PR DESCRIPTION
**Pull request by ell96p**

### CODE REVIEW

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestControllerAdvice
+public class AppExceptionHandler {
+
+    public static final String SOURCE_ERROR_MESSAGE = "Unknown source";
+
+    @ExceptionHandler
+    public ResponseEntity<ErrorResponse> handleException (Exception exc, HttpServletRequest request){
```
Comment (daniel250592) [inline backend/src/main/java/com/crm/exception/AppExceptionHandler.java:17]: final Exception exc,final HttpServletRequest request

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        return new ResponseEntity<>(new ErrorResponse(HttpStatus.BAD_REQUEST.value(),getSource(request),-1, exc.getMessage()),
+        HttpStatus.BAD_REQUEST);
+    }
+
+    private String getSource(HttpServletRequest request) {
```
Comment (daniel250592) [inline backend/src/main/java/com/crm/exception/AppExceptionHandler.java:23]: final

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java (hunk 1 above)
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java:17]: okk

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java (hunk 2 above)
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java:23]: ok

**Review by Fellway — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.servlet.HandlerMapping;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestControllerAdvice
+public class AppExceptionHandler {
+
```
[the remark this answers is not in the document]
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java:13]: ok

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public static final String SOURCE_ERROR_MESSAGE = "Unknown source";
```
[the remark this answers is not in the document]
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java]: ok

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    public ResponseEntity<ErrorResponse> handleException (final Exception exc, final HttpServletRequest request){
+
+        return new ResponseEntity<>(new ErrorResponse(HttpStatus.BAD_REQUEST.value(),getSource(request),-1, exc.getMessage()),
+        HttpStatus.BAD_REQUEST);
+    }
+
+    private String getSource(final HttpServletRequest request) {
+        String source;
+        try {
+            source = request.getAttribute(HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).toString();
```
[the remark this answers is not in the document]
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java]: ok

File: backend/src/main/java/com/crm/exception/AppExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        } catch (Exception e) {
+            source = SOURCE_ERROR_MESSAGE;
```
[the remark this answers is not in the document]
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/AppExceptionHandler.java]: ok

File: backend/src/main/java/com/crm/exception/ErrorResponse.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.crm.exception;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.Getter;
+
+@Getter
+@Setter
+@NoArgsConstructor
```
[the remark this answers is not in the document]
Comment (ell96p) [reply backend/src/main/java/com/crm/exception/ErrorResponse.java]: ok

**Review by Fellway — APPROVED** [review]

**Pull request merged**